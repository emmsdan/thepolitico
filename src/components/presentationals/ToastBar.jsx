/* eslint-disable no-console */
import React from 'react';
import PropTypes from 'prop-types';

const ToastBar = props => {
  let state = props.toast.class;
  return <div className={`toast ${state}`}> {props.toast.message} </div>;
};

ToastBar.propTypes = {
  toast: PropTypes.shape({
    message: PropTypes.string,
    class: PropTypes.string,
  }),
};

export default ToastBar;
