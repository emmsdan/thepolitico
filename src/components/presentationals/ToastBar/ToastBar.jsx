/* eslint-disable no-console */
import React, { useState } from 'react';
import PropTypes from 'prop-types';

const ToastBar = ({ message, className }) => {
  const [ToastClass, setToastClass] = useState('show');
  return (
    <div
      className={`toast ${className} ${ToastClass}`}
      onClick={() => setToastClass('hide')}
    >
      {message}
    </div>
  );
};

ToastBar.propTypes = {
  message: PropTypes.string,
  className: PropTypes.string,
};

export default ToastBar;
