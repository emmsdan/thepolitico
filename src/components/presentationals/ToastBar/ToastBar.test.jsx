import React from 'react';
import { shallow } from 'enzyme';

import ToastBar from './ToastBar';

describe('<ToastBar />', () => {
  const toastbar = {
    message: 'Hi, I am a toastbar',
    className: 'bg-gray',
  };

  it('should render correctly', () => {
    const component = shallow(<ToastBar {...toastbar} />);
    expect(component).toMatchSnapshot();
  });

  it('should display message correctly', () => {
    const component = shallow(<ToastBar {...toastbar} />);
    expect(component).toMatchSnapshot();
    expect(component.find('.toast').text()).toEqual(toastbar.message);
  });

  it('should simulate close toastbar correctly', () => {
    const component = shallow(<ToastBar {...toastbar} />);
    component.find('.toast').simulate('click');
    expect(component.find('.toast.hide').text()).toEqual(toastbar.message);
  });
});
