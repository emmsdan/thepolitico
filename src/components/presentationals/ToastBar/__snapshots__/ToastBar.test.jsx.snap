// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToastBar /> should display message correctly 1`] = `
<div
  className="toast bg-gray show"
  onClick={[Function]}
>
  Hi, I am a toastbar
</div>
`;

exports[`<ToastBar /> should render correctly 1`] = `
<div
  className="toast bg-gray show"
  onClick={[Function]}
>
  Hi, I am a toastbar
</div>
`;
